### PR TITLE
Fish completion

### PR DESCRIPTION
**Pull request by Piraty**
According to the [fish docs](https://fishshell.com/docs/current/#where-to-put-completions), third-party software should install to `/usr/share/fish/vendor_completions.d`.

See: #19413 (rustup: remove fish completion) as well

Some cleanups, fixes and xlints were performed along the way

### CODE REVIEW

File: srcpkgs/ponysay/patches/fish-completion-vendor.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#reason: fish-shell requires 3rd-party completion files to be installed to vendor_completions.d
```
Comment (eli-schwartz) [inline srcpkgs/ponysay/patches/fish-completion-vendor.patch]: There's an upstream patch: https://github.com/erkin/ponysay/pull/298

File: srcpkgs/zathura/patches/fish-completion-vendor.patch [hunk 2]
```diff
@@ -0,0 +1,10 @@
+#reason: fish-shell requires 3rd-party completion files to be installed to vendor_completions.d
```
Comment (eli-schwartz) [inline srcpkgs/zathura/patches/fish-completion-vendor.patch]: There's an upstream patch: https://git.pwmt.org/pwmt/zathura/commit/d21d0406630eaf5e20d180524e6d396611e7243a

(I have no idea why I cannot find my Merge Request on their gitlab instance anymore. :/)

**Comment by travankor** [comment]
ripgrep had its completion removed in [ca7e4a44](https://github.com/void-linux/void-packages/pull/19096), so could add that one back

**Comment by Johnnynator** [comment]
> ripgrep had its completion removed in [ca7e4a44](https://github.com/void-linux/void-packages/pull/19096), so could add that one back

No, the completion is provided by fish itself, no point in shipping it twice. (And the completions are the same)

**Comment by travankor** [comment]
umm, https://github.com/fish-shell/fish-shell/commit/f036d019618ff1f0508c539dc88d64a194cd2034#diff-6cd1d7d06e00138db86e13bde99ed616

**Comment by travankor** [comment]
https://github.com/fish-shell/fish-shell/commit/f036d019618ff1f0508c539dc88d64a194cd2034#diff-6cd1d7d06e00138db86e13bde99ed616

**Comment by Johnnynator** [comment]
> [fish-shell/fish-shell@f036d01#diff-6cd1d7d06e00138db86e13bde99ed616](https://github.com/fish-shell/fish-shell/commit/f036d019618ff1f0508c539dc88d64a194cd2034#diff-6cd1d7d06e00138db86e13bde99ed616)

Thx for pointing this out. Since it seems to be dropped with the next release, it makes sense to readd the completions.

**Comment by Piraty** [comment]
thanks for feedback. anything else ?

**Comment by Johnnynator** [comment]
lgtm

**Comment by Piraty** [comment]
:shipit:

**Pull request merged**